src/manage_nsfs/manage_nsfs_validations.js

```diff
@@ -242,19 +243,36 @@ function validate_flags_combination(type, action, input_options) {
 }
 
 /**
- * validate_flags_value_combination checks flags and value combination.
- * 1. account add or update - name should not be anonymous
+ * validate_account_name
+ * We check the name only on new accounts (account add) or accounts' rename (account update)
+ * In case, we had already an account with invalid name, it can be changed to a valid name
+ * (current name is not validated)
  * @param {string} type
  * @param {string} action
  * @param {object} input_options_with_data
  */
-function validate_flags_value_combination(type, action, input_options_with_data) {
+function validate_account_name(type, action, input_options_with_data) {
     if (type === TYPES.ACCOUNT) {
-        if (action === ACTIONS.ADD || action === ACTIONS.UPDATE) {
-            if (input_options_with_data.name === ANONYMOUS || input_options_with_data.new_name === ANONYMOUS) {
-                const detail = 'Account name \'anonymous\' is not valid';
-                throw_cli_error(ManageCLIError.InvalidAccountName, detail);
+        // in name and new_name we allow type number, hence convert it to string
+        // (it is saved converted in fetch_account_data)
+        let account_name;
+        try {
+            if (action === ACTIONS.ADD) {
+                account_name = String(input_options_with_data.name);
+                iam_utils.validate_username(account_name, 'name');
+            } else if (action === ACTIONS.UPDATE && input_options_with_data.new_name !== undefined) {
```

Comment:
add specific case for not allowing update of anonymous account using this condition